### ISSUE TITLE
Opening universe described in HBondAnalysis docs causes error

### ISSUE THREAD

**Issue by jdetle** [MDAnalysis/mdanalysis#838]
### Expected behaviour
Running HBondAnalysis as specified in the docs should work as expected, if I am missing something simple please let me know. 

### Actual behaviour 
![error](https://cloud.githubusercontent.com/assets/10944582/14946553/a53774de-0fd8-11e6-9674-eba59fdc4fde.jpg)

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
u = mda.Universe(PSF, PDB)
....
```

### Current version of MDAnalysis:
0.14.1-dev0

**Comment by kain88-de** [comment]
Do `PSF` and `PDB` contain the same protein? It looks like we are trying to read more atoms from the PDB then we actually allocated from the PSF topology.

**Comment by richardjgowers** [comment]
Yeah the PDB file is much longer (it's solvated), I think PSF DCD is the correct combination

**Comment by jdetle** [comment]
Yea this is an error in the Docs, I'll go ahead and change PDB to DCD. This will be solved by #833.

**Comment by kain88-de** [comment]
Thanks.

**Comment by orbeckst** [comment]
>  I think PSF DCD is the correct combination

@jdetle also noted that the [example](http://pythonhosted.org/MDAnalysis/documentation_pages/analysis/hbond_analysis.html#detection-of-hydrogen-bonds) wrongly suggests you could use analyze water-protein hydrogen bonds. But the PSF/DCD combo is in implicit solvent and does *not* have TIP3 water inside.

Instead use something like the following example where we calculated the average number of hydrogen bond for protein residues that are solvent-exposed:

```python
import numpy as np
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import GRO, XTC
from MDAnalysis.analysis.hbonds import HydrogenBondAnalysis

u = mda.Universe(GRO, XTC)

# hydrogen bonds
H = HydrogenBondAnalysis(u, 'protein', 'resname SOL', distance=3, angle=120)
H.run()
counts = H.count_by_time()
print(counts.count)
# [586 594 577 588 573 610 599 596 598 602]

# exposed residues (quick to write, slow to execute...)
Nres_exposed = np.array([u.select_atoms("protein and around 3 resname SOL").n_residues for ts in u.trajectory], dtype=np.float64)

# hbonds per residue
avg_hbonds = counts.count/Nres_exposed

# plot
import matplotlib.pyplot as plt
plt.matplotlib.style.use('ggplot')
fig = plt.figure(figsize=(5,5))
ax = fig.add_subplot(111)
ax.fill_between(counts.time, avg_hbonds, color="blue", alpha=0.3)
ax.set_xlabel(r"time $t$ (ps)")
ax.set_ylabel(r"number of hydrogen bonds protein-water $N$")
fig.savefig("adk_avg_protein_water_hbonds.png")
```

![adk_avg_protein_water_hbonds](https://cloud.githubusercontent.com/assets/237980/15002412/c76e0b3e-1155-11e6-8504-528aba5a24ba.png)

**Comment by orbeckst** [comment]
@jdetle, will this be fixed with PR #833?

**Comment by jdetle** [comment]
Yep!
On Fri, May 6, 2016 at 2:07 PM Oliver Beckstein <notifications@github.com>
wrote:

> @jdetle <https://github.com/jdetle>, will this be fixed with PR #833
> <https://github.com/MDAnalysis/mdanalysis/pull/833>?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/MDAnalysis/mdanalysis/issues/838#issuecomment-217558725>
>
-- 
Have a wonderful day,
John J. Detlefs
